src/ext/conversion/importProperties.ts

```diff
@@ -18,7 +54,30 @@ const importProperties = (
       defaultMeasure: extension.getDefaultMeasureProperties(),
     },
     hypercubePath,
+    extension,
   });
+  const {
+    qHyperCubeDef: { qColumnOrder, columnWidths },
+  } = propertyTree.qProperty;
+  let {
+    qHyperCubeDef: { qDimensions, qMeasures },
+  } = propertyTree.qProperty;
+  const numDims = qDimensions.length;
+  const columnsLength = numDims + qMeasures.length;
+  const columnOrder = qColumnOrder?.length === columnsLength ? qColumnOrder : Array.from(Array(columnsLength).keys());
+
+  qDimensions = columnOrder
+    .filter((colIdx: number) => colIdx < numDims)
+    .map((colIdx: number) => getColumnInfo(qDimensions, colIdx, columnWidths));
+
+  qMeasures = columnOrder
+    .filter((colIdx: number) => colIdx >= numDims)
+    .map((colIdx: number) => getColumnInfo(qMeasures, colIdx, columnWidths, numDims));
```

Comment:
you could just pass `colIdx - numdims` and you don't have to pass numDims and have logic for that in `getColumnInfo`

Reply:
either have `colIdx - numdims` in parameters or inside function. I prefer to have that inside function

```
getColumnInfo = (
  columnInfo: DimensionProperties[] | MeasureProperties[],
  colIdx: number,
  columnWidths?: number[],
  measureIndex?: number
) => {
  let index = colIdx;
  if (measureIndex) index = measureIndex;
  const column = columnInfo[index];
...
```

Reply:
well that makes less sense to me, if you want to have logic for it, fine. But at least pass only one index. So leave it as it is if you want